docs/source/howto/codes/interfacing.rst

```diff
@@ -0,0 +1,282 @@
+.. _how-to:codes:plugin:
+
+Interfacing external codes
+==========================
+
+In order to run an external code, you will first have to explain to AiiDA how to run it: how to create inputs files from the content of the input node, how the program can read those input files, where to find and how to read the output files, etc.
+The way you transmit these instructions to AiiDA is by extending the |CalcJob| and the |Parser| classes:
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+    from aiida.engine import CalcJob
+
+    class ArithmeticAddCalculation(CalcJob):
+
+        @classmethod
+        def define(cls, spec):
+            super().define(spec)
+            # Specifications of inputs, outputs and exit codes.
+            # no return statement
+
+        def prepare_for_submission(self, folder):
+            # Instructions on how to prepare the input files
+            return calcinfo
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+    from aiida.parsers.parser import Parser
+
+    class ArithmeticAddParser(Parser):
+
+        def parse(self, **kwargs):
+            # Instructions on how to process the output files
+            # no return statement
+
+As you can guess from the name of the child-classes in the previous code snippet, we will illustrate the implementation of this code wrapping procedure with a possible design for the example introduced in the :ref:`tutorial<tutorial:basic>` of a code that sums two numbers.
```

Comment:
I think it would be better to not directly reference the tutorial. For the how-to's, we make the assumption that people know the things we teach in the tutorial, but the how-to's are not a direct extension of the tutorial.

Think of it this way, an AiiDA expert might still want to look some things up among the how-to guides, but may never have gone through the tutorial.

Reply:
Ok; I think it still may be relevant if someone who went through the tutorial to place themselves more easily, but I reduced the presence to a "btw" comment that can be just ignored by more expert users:

> As you may have guessed from the name of the child-classes in the previous code snippet, we will illustrate this code wrapping procedure by designing a calculation that will add up two numbers (note that this is the same example used in the `tutorial`).

Would this be better or do you still think is best to remove it completely?

Reply:
Yes, I think that's good.